src/nodes/file.rs

```diff
@@ -91,4 +99,25 @@ impl Node for File {
 
         Ok(state.attr)
     }
+
+    fn open(&self, flags: u32) -> NodeResult<Arc<Handle>> {
+        let state = self.state.lock().unwrap();
+
+        let flags = flags as i32;
+        let mut options = fs::OpenOptions::new();
+        options.read(true);
+        if flags & (libc::O_WRONLY | libc::O_RDWR) != 0 {
+            if !self.writable {
+                return Err(KernelError::from_errno(libc::EACCES));
+            }
+            if flags & libc::O_WRONLY != 0 {
+                options.read(false);
+            }
+            options.write(true);
+        }
+        options.custom_flags(flags);
+
+        let file = options.open(state.underlying_path.as_ref().unwrap())?;
```

Comment:
How can you be sure underlying_path is Some here?

Reply:
Oops, good catch. Files and directories needn't have an optional underlying path... but I was led the wrong path when trying to share the implementation of `getattr`. Removed optional now.